### DISCUSSION TITLE
Customize font of Sync Dirs Window

### DISCUSSION THREAD

**Discussion by lars18th** [doublecmd/doublecmd#1508] [Suggestions]
Hi,

I recommend that the content of the Sync Dirs uses the configured fonts. For example, it could use the "Main Font" or the "Search Results Font" (or anything else). The current used font is ugly in some environments. And it really doesn't have sense that some font can't be changed with the large list of Fonts to configure.

Please, consider this enhancement. I don't suggest to add any new entry in the configuration; only to link the font used to any of the defined in the current list.

Thank you.

**Comment by lars18th** [comment]
FYI, I've already implemented this in PR #1509 .

**Comment by lars18th** [comment]
Only FYI I'm compiling last releases with this patch #1509. I recommend to do it until the main developer wants to consider a custom font for the Sync Dirs window. The default is _very_ ugly in Windows with a HDPI monitor.

**Comment by Skif-off** [comment]
> I recommend to do it

This is ~~very ugly~~ not a pretty hack. Maybe something like this
```diff
--- src/fsyncdirsdlg.pas
+++ src/fsyncdirsdlg.pas
@@ -897,6 +897,7 @@
   x: Integer;
   s: string;
 begin
+  FontOptionsToFont(gFonts[dcfSyncDirs], Canvas.Font);
   if (FVisibleItems = nil) or (aRow >= FVisibleItems.Count) then Exit;
   with MainDrawGrid.Canvas do
   begin
--- src/uglobs.pas
+++ src/uglobs.pas
@@ -121,7 +121,7 @@
   TPluginType = (ptDSX, ptWCX, ptWDX, ptWFX, ptWLX); //*Important: Keep that order to to fit with procedures LoadXmlConfig/SaveXmlConfig when we save/restore widths of "TfrmTweakPlugin".
   TWcxCfgViewMode = (wcvmByPlugin, wcvmByExtension);
 
-  TDCFont = (dcfMain, dcfEditor, dcfViewer, dcfViewerBook, dcfLog, dcfConsole, dcfPathEdit, dcfSearchResults, dcfFunctionButtons, dcfTreeViewMenu, dcfStatusBar);
+  TDCFont = (dcfMain, dcfEditor, dcfViewer, dcfViewerBook, dcfLog, dcfConsole, dcfPathEdit, dcfSearchResults, dcfFunctionButtons, dcfTreeViewMenu, dcfStatusBar, dcfSyncDirs);
   TDCFontOptions = record
     Usage: string;
     Name: string;
@@ -1840,6 +1840,13 @@
   gFonts[dcfStatusBar].MinValue := 6;
   gFonts[dcfStatusBar].MaxValue := 200;
 
+  gFonts[dcfSyncDirs].Name := 'default';
+  gFonts[dcfSyncDirs].Size := 10;
+  gFonts[dcfSyncDirs].Style := [fsBold];
+  gFonts[dcfSyncDirs].Quality := fqDefault;
+  gFonts[dcfSyncDirs].MinValue := 6;
+  gFonts[dcfSyncDirs].MaxValue := 200;
+
   { Colors page }
   gUseCursorBorder := False;
   gUseFrameCursor := False;
@@ -2714,6 +2721,7 @@
     if LoadedConfigVersion >= 11 then GetDCFont(gConfig.FindNode(Root, 'Fonts/SearchResults'), gFonts[dcfSearchResults]); //Let's ignore possible previous setting for this and keep our default.
     GetDCFont(gConfig.FindNode(Root, 'Fonts/TreeViewMenu'), gFonts[dcfTreeViewMenu]);
     GetDCFont(gConfig.FindNode(Root, 'Fonts/StatusBar'), gFonts[dcfStatusBar]);
+    GetDCFont(gConfig.FindNode(Root, 'Fonts/SyncDirs'), gFonts[dcfSyncDirs]);
 
     { Colors page }
     Node := Root.FindNode('Colors');
@@ -3428,6 +3436,7 @@
     SetDCFont(gConfig.FindNode(Root, 'Fonts/SearchResults',True), gFonts[dcfSearchResults]);
     SetDCFont(gConfig.FindNode(Root, 'Fonts/TreeViewMenu', True), gFonts[dcfTreeViewMenu]);
     SetDCFont(gConfig.FindNode(Root, 'Fonts/StatusBar', True), gFonts[dcfStatusBar]);
+    SetDCFont(gConfig.FindNode(Root, 'Fonts/SyncDirs', True), gFonts[dcfSyncDirs]);
 
     { Colors page }
     Node := FindNode(Root, 'Colors', True);
```
?
But I see `Font.Bold := True;` below and it makes me sad because I'm not a programmer.

**Reply by lars18th** [reply]
Thank you. And yes, why you use `gFonts[dcfSyncDirs].Style := [fsBold];`? It could be `gFonts[dcfSyncDirs].Style := [];`

**Reply by Skif-off** [reply]
> It could be `gFonts[dcfSyncDirs].Style := [];`

Why? This patch allows to manually specify the desired font (in theory), but by default, this font is _bold_.

I think this patch needs refinement, see [src/fsyncdirsdlg.pas#L900](https://github.com/doublecmd/doublecmd/blob/fefa77a1b0a6fbc1f943151dfb4e9101b0f7a952/src/fsyncdirsdlg.pas#L900) and below:
```pas
Font.Bold := True;
```
it looks like a forced switching and in this case this line must be removed, but it can be important (I don't know).

**Comment by lars18th** [comment]
Hi @Skif-off ,

Why not create a PR with this?

**Comment by Skif-off** [comment]
I don't see the point, @alexx2000 has already expressed his [opinion](https://github.com/doublecmd/doublecmd/pull/1509#issuecomment-2791771936).
